Explorer/Assets/DCL/InWorldCamera/ReelActions/ReelCommonActions.cs

```diff
@@ -29,11 +37,44 @@ public static void CopyReelLink(string reelId, IDecentralandUrlsSource decentral
         }
 
         /// <summary>
-        ///     Opens a browser tab to the full resolution reel url
+        ///     Downloads a reel image to local storage in {home_directory}/{DECENTRALAND_REELS_HOME_FOLDER}/{reelId}
+        ///     and opens the default file browser at that location
         /// </summary>
-        public static void DownloadReel(string reelUrl, IWebBrowser webBrowser)
+        public static async UniTask DownloadReelToFileAsync(string reelUrl, CancellationToken ct)
+        {
+            using (UnityWebRequest webRequest = UnityWebRequestTexture.GetTexture(reelUrl))
+            {
+                Uri uri = new Uri(reelUrl);
+                await webRequest.SendWebRequest().ToUniTask(cancellationToken: ct);
+
+                if (webRequest.result != UnityWebRequest.Result.Success)
+                    throw new Exception($"Error while downloading reel: {webRequest.error}");
+
+                Texture2D texture = DownloadHandlerTexture.GetContent(webRequest);
+                byte[] imageBytes = texture.EncodeToPNG();
+                string pathToReelsFolder = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), DECENTRALAND_REELS_HOME_FOLDER);
+                string absolutePath = Path.Combine(pathToReelsFolder, Path.GetFileName(uri.LocalPath));
```

Comment:
Path combine allocates memory each time, using StringBuilder would be better to also allow reusal of the StringBuilder